### PR TITLE
Development

### PR DESCRIPTION
**Pull request by SHA888**
- Added user pallet
- Added treasury pallet

### CODE REVIEW

File: pallets/users/src/lib.rs [hunk 1]
```diff
@@ -32,60 +35,83 @@ pub mod pallet {
 
 	#[pallet::storage]
 	#[pallet::getter(fn something)]
-
+	// define pallet storage abilities up top and create function definitions inside
 	pub type Something<T> = StorageValue<_, u32>;
```
Comment (SHA888) [inline pallets/users/src/lib.rs:39]: Need to update this. The rest is good.

File: pallets/users/src/lib.rs [hunk 2]
```diff
@@ -32,60 +35,83 @@ pub mod pallet {
...
+			ensure!(!Users::<T>::contains_key(&who), Error::<T>::UserAlreadyExists);
+			<Users<T>>::insert(&who, User { rank_points: 0 });
+
+			Self::deposit_event(Event::UserCreated( who));
+
+			Ok(())
+		}
 
 		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
 		pub fn cause_error(origin: OriginFor<T>) -> DispatchResult {
```
Comment (SHA888) [inline pallets/users/src/lib.rs:104]: I don't think we need this, right?

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -295,6 +335,114 @@ impl pallet_sudo::Config for Runtime {
 impl pallet_chocolate::Config for Runtime {
 	type Event = Event;
 }
+/// Configure the pallet-users in pallets/users.
+impl pallet_users::Config for Runtime {
```
Comment (SHA888) [inline runtime/src/lib.rs:339]: Shoudn't we need to add something more here?

**Review by SHA888 — COMMENTED** [review]
@islami00 Could you make some changes / review / discussion?

File: pallets/users/src/lib.rs (hunk 1 above)
Comment (islami00) [reply pallets/users/src/lib.rs]: Updated. That part is boilerplate so I simply removed it.
Fn, User only contains `rank_points`

**Comment by SHA888** [comment]
Looks great. Please proceed to merge @islami00

File: pallets/users/src/lib.rs (hunk 2 above)
Comment (islami00) [reply pallets/users/src/lib.rs]: Yeah, I'll change it when there's power. I think a more efficient way would be to create records on the fly in the same dispatch

File: runtime/src/lib.rs (hunk 3 above)
Comment (islami00) [reply runtime/src/lib.rs:339]: Not necessarily, the types defined in the chocolate module are simply aliases,  hence the y don't need any external configuration.

**Pull request merged**